pkg/gen/python-templates/CustomResource.py

```diff
@@ -0,0 +1,49 @@
+# *** WARNING: this file was generated by the Pulumi Kubernetes codegen tool. ***
+# *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import pulumi
+import pulumi.runtime
+
+from .. import tables
+
+
+class CustomResource(pulumi.CustomResource):
+    """
+    CustomResource represents an instance of a CustomResourceDefinition (CRD). For example, the
+    CoreOS Prometheus operator exposes a CRD `monitoring.coreos.com/ServiceMonitor`; to
+    instantiate this as a Pulumi resource, one could call `new CustomResource`, passing the
+    `ServiceMonitor` resource definition as an argument.
+    """
+
+    def __init__(self, __name__, __opts__=None, api_version=None, kind=None, metadata=None, spec=None):
+        if not __name__:
+            raise TypeError('Missing resource name argument (for URN creation)')
+        if not isinstance(__name__, str):
+            raise TypeError('Expected resource name to be a string')
+        if __opts__ and not isinstance(__opts__, pulumi.ResourceOptions):
+            raise TypeError('Expected resource options to be a ResourceOptions instance')
+
+        __props__ = dict()
+
+        if not api_version:
+            raise TypeError('Missing required property apiVersion')
+        __props__['apiVersion'] = api_version
+        if not kind:
+            raise TypeError('Missing required property kind')
+        __props__['kind'] = kind
+        if spec is None:
+            raise TypeError('Missing required property spec')
+        __props__['spec'] = spec
+        __props__['metadata'] = metadata
```

Comment:
is it necessary for us to validate that `metadata` is not `None`? i know we set the `metadata.name` in pulumi, which is the only real req out of `metadata`, but i suppose we don't check it's `None` because this value can be valid if there aren't any other fields other than the `name`? is that your thinking too?

Reply:
Right, metadata isn't required, so `None` is acceptable.